python/ELASTICACHE_REDIS_CLUSTER_AUTO_BACKUP_CHECK/ELASTICACHE_REDIS_CLUSTER_AUTO_BACKUP_CHECK.py

```diff
@@ -0,0 +1,444 @@
+        else:
+            cache_clusters_result = ec_client.describe_cache_clusters(Marker=marker, MaxRecords=100, ShowCacheNodeInfo=False, ShowCacheClustersNotInReplicationGroups=True)
+        cache_clusters.extend(cache_clusters_result['CacheClusters'])
+        if 'Marker' in cache_clusters_result:
+            marker = cache_clusters_result['Marker']
+        else:
+            return cache_clusters
+
+
+def generate_evaluations(cache_clusters, replication_groups, snapshot_retention_period, event):
+    evaluations = []
+    if cache_clusters:
+        for cluster in cache_clusters:
+            if cluster['Engine'] == 'redis':
+                if cluster['SnapshotRetentionLimit'] == 0:
+                    evaluations.append(build_evaluation(cluster['CacheClusterId'], 'NON_COMPLIANT', event, resource_type='AWS::ElastiCache::CacheCluster', annotation="Automatic backup not enabled for Amazon ElastiCache cluster: {}".format(cluster['CacheClusterId'])))
+                elif cluster['SnapshotRetentionLimit'] < snapshot_retention_period:
+                    evaluations.append(build_evaluation(cluster['CacheClusterId'], 'NON_COMPLIANT', event, resource_type='AWS::ElastiCache::CacheCluster', annotation='Automatic backup retention period for Amazon ElastiCache cluster {} is less then {} day(s).'.format(cluster['CacheClusterId'], snapshot_retention_period)))
+                else:
+                    evaluations.append(build_evaluation(cluster['CacheClusterId'], 'COMPLIANT', event, resource_type='AWS::ElastiCache::CacheCluster'))
+    if replication_groups:
+        for replication_group in replication_groups:
+            if replication_group['SnapshotRetentionLimit'] < snapshot_retention_period:
+                evaluations.append(build_evaluation(replication_group['ReplicationGroupId'], 'NON_COMPLIANT', event, resource_type='AWS::ElastiCache::CacheCluster', annotation='Automatic backup retention period for Amazon ElastiCache cluster {} is less then {} day(s).'.format(replication_group['ReplicationGroupId'], snapshot_retention_period)))
+            elif replication_group['SnapshotRetentionLimit'] == 0:
+                evaluations.append(build_evaluation(replication_group['ReplicationGroupId'], 'NON_COMPLIANT', event, resource_type='AWS::ElastiCache::CacheCluster', annotation="Automatic backup not enabled for Amazon ElastiCache cluster: {}".format(replication_group['ReplicationGroupId'])))
+            else:
+                evaluations.append(build_evaluation(replication_group['ReplicationGroupId'], 'COMPLIANT', event, resource_type='AWS::ElastiCache::CacheCluster'))
+    return evaluations
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    ec_client = get_client('elasticache', event)
+    cache_clusters = get_cache_clusters(ec_client)
+    replication_groups = get_replication_groups(ec_client)
+    if not cache_clusters and not replication_groups:

```

Comment:
Can you make this decision before calling all the API?

Does `if not cache_clusters:` allows you to make a decision
What about `if not replication_groups:`

The goal is to reduce the number of API calls in all situations.